workspaces/optic-domain/src/async-tools.ts

```diff
@@ -37,6 +38,22 @@ export function lastBy<T>(
   };
 }
 
+export async function* intoJSONArray<T>(
```

Comment:
I tried looking for an existing function that did this for me, but I couldn't find anything

Reply:
`stream-json` doesn't only have parsers, but also what they call [stringers](https://github.com/uhop/stream-json/wiki/Stringer) (for both json and jsonl, see docs), which are generally used in combination with a [Disassembler](https://github.com/uhop/stream-json/wiki/Disassembler). The dissambler accepts entities and returns a token stream, which the stringer uses to emit JSON. Even the biggest entities can be smoothly rendered as JSON that way!

Reply:
I updated this function to return a `Readable` stream - since this is likely it be used to serialize back into a IO using a stream. I could update this to return an async iterable but I'm not sure what usecase this would be necessary